docs/troubleshooting.md

```diff
@@ -0,0 +1,69 @@
+    
+Next, find the cloud env var for your cluster. You can find instructions [here](https://docs.microsoft.com/en-us/azure/storage/common/storage-powershell-independent-clouds#get-endpoint-using-get-azenvironment). 
+
+Once you have that, create the `azureoperatorsettings.yaml` file from step 6 mentioned [here](https://operatorhub.io/operator/azure-service-operator), and replace the placeholder values into the respective fields. Finally, run `kubectl apply -f <location of your azureoperatorsettings.yaml file>`.
+
+You should be able to see `azureoperator-controller-manager-***` pods running in your cluster. To view the pods, run `kubectl get pods -n operators` 
+
+## I have already installed `azureserviceoperator` but I'm not sure which service principal secret is in use?
+This command might be able to help you see your secret locally. 
+```kubectl get secret azureoperatorsettings -n operators -o go-template='{{range $k,$v := .data}}{{printf \"%s: \" $k}}{{if not $v}}{{$v}}{{else}}{{$v | base64decode}}{{end}}{{\"\n\"}}{{end}}'```
+[SO thread for reference](https://stackoverflow.com/questions/56909180/decoding-kubernetes-secret/58117444#58117444)
+
+## What if I see the operator namespace already exists for the cluster?
+
+If you see that the operator namespace already exists for the cluster, it's likely the result of trying to install the operator previously. It also means that you can now successfully move on to the next step. 
+However, this can also be a symptom of incorrectly installing the operator, for example, trying to run the install guide out of order. Other common issues related to an "out-of-order install":
+
+* `namespace already exist` - This means that azure-service-operator (ASO) has already been run for the cluster before.
+* `service-principal` issues: If you're seeing silent failures with service principal in use, meaning it is expired or needs to be reset, you can follow the steps as mentioned [here](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli#create-a-service-principal) or in step 4 here in the [OperatorHub guidance](https://operatorhub.io/operator/azure-service-operator).
+* `kubectl api-resource` failure with `error: unable to retrieve the complete list of server APIs: metcs.k8s.io/vabeta=1: the server is currently unable to handle the request.` : This error indicates a networking error within your cluster and is not related to your Azure Service Operator installation or configuration.
+
+ ## When trying to install cert-manager, I'm seeing that the connection is refused for `cert-manager-package`?
+This could be related to following open error: https://github.com/jetstack/cert-manager/issues/2752
+ 
+ ## The azureoperator-controller-manager seems to be in a crash loop. How do I investigate?
+
+`kubectl describe  pod operators`
+
+To get logs in case ASO operator is crashlooping : `kubectl logs -n  operators pod/azureoperator-controller-manager-****** --container manager`
```

Comment:
you might want to explain what the user should reaplace -****** with...

Reply:
cool thanks will try to come up with something and then you or @sakthi-vetrivel can correct it. thanks.